modules/storages/app/workers/storages/manage_nextcloud_integration_events_job.rb

```diff
@@ -30,22 +30,23 @@ module Storages
   class ManageNextcloudIntegrationEventsJob < ApplicationJob
     include ManageNextcloudIntegrationJobMixin
 
-    DEBOUNCE_TIME = 5.seconds.freeze
+    THREAD_DEBOUNCE_TIME = 4.seconds.freeze
+    JOB_DEBOUNCE_TIME = 5.seconds.freeze
     THREAD_KEY = :manage_nextcloud_integration_events_job_debounce_happend_at
 
     queue_with_priority :above_normal
 
     def self.debounce
-      last_debounce_happend_at = Thread.current[THREAD_KEY]
-      if last_debounce_happend_at.blank? || Time.current > (last_debounce_happend_at + DEBOUNCE_TIME)
+      last_debounce_happend_at = RequestStore.store[THREAD_KEY]
+      if last_debounce_happend_at.blank? || Time.current > (last_debounce_happend_at + THREAD_DEBOUNCE_TIME)
```

Comment:
Again `THREAD` terminology. `THREAD_DEBOUNCE_TIME`

Reply:
I think in this case the name is all right. Check comment above, please.